From f16b95fe0764e6a99cc44062a1d29d3b3c1d9756 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 15 Mar 2021 23:20:51 +0200
Subject: [PATCH] [SW-28697] boot_if: cpu_boot_cmd_get() return u32

Since we are going to use this API in both, new comms framework, and the
legacy LKD protocol, generalize the return value and make it u32.
This removes the ambiguous conversions.

Change-Id: Ibe95797657037fff1c6c2bebf917468f398dd939
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/boot_if.c b/common/boot_if.c
index 6b4f424..3dcaa7f 100644
--- a/common/boot_if.c
+++ b/common/boot_if.c
@@ -27,7 +27,7 @@
 	regmap_writel(KMD_MSG_NA, mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU);
 }
 
-enum kmd_msg cpu_boot_cmd_get(void)
+u32 cpu_boot_cmd_get(void)
 {
 	return regmap_readl(mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU);
 }
diff --git a/include/boot_if.h b/include/boot_if.h
index 1e12747..59a26ed 100644
--- a/include/boot_if.h
+++ b/include/boot_if.h
@@ -10,7 +10,7 @@
 #include <hl_boot_if.h>
 #include <preboot.h>
 
-enum kmd_msg cpu_boot_cmd_get(void);
+u32 cpu_boot_cmd_get(void);
 void cpu_boot_cmd_clr_before_wfe(void);
 void cpu_boot_sts_set(u32 val);
 void cpu_boot_err_clrset(u32 mask, bool set);
